packages/jest-core/src/cli/index.ts

```diff
@@ -167,13 +155,28 @@ const buildContextsAndHasteMaps = async (
   return {contexts, hasteMapInstances};
 };
 
-const _run10000 = async (
+/**
+ * Runs Jest either in watch mode or as a one-off. This is a lower-level API than `runCLI` and is intended for internal use by `runCLI` or externally.
+ * Note that `process.exit` might be called when using `globalConfig.watch` or `globalConfig.watchAll` is true.
+ *
+ * @param globalConfig The global configuration to use for this run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param configs The project configurations to run. It can be obtained using `readConfigs` (imported from 'jest-config').
+ * @param warnForDeprecations Whether or not to warn for deprecation messages when `globalConfig.watch` or `globalConfig.watchAll` is true.
+ * @param outputStream The stream to write output to. If not provided, it defaults to `process.stdout`.
+ * @returns A Promise that resolves to the result, or never resolves when `globalConfig.watch` or `globalConfig.watchAll` is true.
+ * @example
+ * import { runCore, readConfigs } from 'jest';
+ *
+ * const { globalConfig, configs } = await readConfigs(process.argv, [process.cwd()]);
```

Comment:
Should follow the same style as else where, perhaps?

```suggestion
 * import {runCore, readConfigs} from 'jest';
 *
 * const {globalConfig, configs} = await readConfigs(process.argv, [process.cwd()]);
```